src/compiler/checker.ts

```diff
@@ -7596,6 +7596,9 @@ namespace ts {
                 }
                 return anyType;
             }
+            if (containsThisProperty(expression.left, expression.right)) {

```

Comment:
```suggestion
            if (containsSameNamedThisProperty(expression.left, expression.right)) {
```

(this PR is a bit drafty so I didn't take time to create good names)